addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAccessoryRegistry.java

```diff
@@ -0,0 +1,102 @@
+package org.openhab.io.homekit.internal;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.openhab.io.homekit.internal.accessories.GroupedAccessory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.beowulfe.hap.HomekitAccessory;
+import com.beowulfe.hap.HomekitRoot;
+
+/**
+ * Stores the created HomekitAccessories. GroupedAccessories are also held here
+ * in a pre-created pending state until all required characteristics are found.
+ *
+ * @author Andy Lintner
+ */
+class HomekitAccessoryRegistry {
+
+    private HomekitRoot bridge;
+    private final List<HomekitAccessory> createdAccessories = new LinkedList<>();
+    private final Set<Integer> createdIds = new HashSet<>();
+    private final Map<String, GroupedAccessory> pendingGroupedAccessories = new HashMap<>();
+    private final List<HomekitTaggedItem> pendingCharacteristics = new LinkedList<>();
+
+    private final Logger logger = LoggerFactory.getLogger(HomekitAccessoryRegistry.class);
+
+    public void remove(HomekitTaggedItem taggedItem) {
+        Iterator<HomekitAccessory> i = createdAccessories.iterator();
+        while (i.hasNext()) {
+            HomekitAccessory accessory = i.next();
+            if (accessory.getId() == taggedItem.getId()) {
+                logger.info("Removed accessory " + accessory.getId());
```

Comment:
I'd prefer to have most logging statements (here and below in the code) on debug level only - keep the info level for stuff that the normal user should really see during normal operation.